app/main.py

```diff
@@ -1,2 +1,77 @@
+from typing import Hashable, Any
+
+DEFAULT_CAPACITY = 8
+
+
+class Node:
+    def __init__(self, key: Hashable, value: Any) -> None:
+        self.key = key
+        self.value = value
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.capacity = DEFAULT_CAPACITY
+        self.hash_table = [None] * self.capacity
+        self.number_of_elements = 0
+
+    def __len__(self) -> int:
+        return self.number_of_elements
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        if self.number_of_elements >= self.capacity * 0.75:
+            self.resize()
+
+        hash_key = hash(key)
+        index_to_past = self.find_available_case(key, hash_key)
+
+        while (
```

Comment:
this block of code repeats in both set and get method + you have find_available_case method. Separate logic of finding correct index in a separate method and reuse it to keep your code DRY.